### PR TITLE
chore: add port to connection details

### PR DESCRIPTION
**Pull request by aiell0**
### Description of your changes

Fixes https://github.com/crossplane-contrib/provider-upjet-aws/issues/1458

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Tested with [uptest](https://github.com/crossplane-contrib/provider-upjet-aws/actions/runs/10501480232) and manually [commented](https://github.com/crossplane-contrib/provider-upjet-aws/pull/1460#issuecomment-2303615597) below

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by aiell0** [comment]
/test-examples="examples/opensearch/v1beta1/domain.yaml"

**Comment by aiell0** [comment]
/test-examples="examples/opensearch/v1beta2/domain.yaml"

File: config/elasticache/config.go [hunk 1]
```diff
@@ -33,6 +33,9 @@ func Configure(p *config.Provider) { //nolint:gocyclo
 			if a, ok := attr["cluster_address"].(string); ok {
 				conn["cluster_address"] = []byte(a)
 			}
+			if a, ok := attr["port"].(string); ok {
```
Comment (turkenf) [inline config/elasticache/config.go:36]: I don't see the port attribute directly in the terraform documentation of this resource, did you test it manually? I think we need to test `examples/elasticache/v1beta1/cluster.yaml`.

**Comment by aiell0** [comment]
/test-examples="examples/elasticache/v1beta1/cluster.yaml"

File: config/elasticache/config.go (hunk 1 above)
Comment (aiell0) [reply config/elasticache/config.go:36]: I have not tested manually, it's not in the Terraform docs but it is in the schema. And you are right, I ran the wrong tests, running correct ones now.

**Comment by aiell0** [comment]
/test-examples="examples/elasticache/v1beta2/cluster.yaml"

**Comment by aiell0** [comment]
/test-examples="examples/elasticache/v1beta1/cluster.yaml"

**Comment by turkenf** [comment]
Thank you for your effort, @aiell0; it would be great if you could update the description to show that you are testing manually and seeing the port value in the secret.

**Comment by aiell0** [comment]
@turkenf I had to change the code to reflect the correct type of port (number vs string) and it worked! Result is below:

```
lawrenceaiello@Lawrences-MacBook-Pro ~/Development/aiell0/provider-upjet-aws (add-port-to-elasticache-connection-details) $ k get secrets                                                                                                 <aws:dev>
NAME             TYPE                                DATA   AGE
sample-cluster   connection.crossplane.io/v1alpha1   2      7m14s
lawrenceaiello@Lawrences-MacBook-Pro ~/Development/aiell0/provider-upjet-aws (add-port-to-elasticache-connection-details) $ k describe secrets                                                                                            <aws:dev>
Name:         sample-cluster
Namespace:    default
Labels:       <none>
Annotations:  <none>

Type:  connection.crossplane.io/v1alpha1

Data
====
cluster_address:  50 bytes
port:             5 bytes
lawrenceaiello@Lawrences-MacBook-Pro ~/Development/aiell0/provider-upjet-aws (add-port-to-elasticache-connection-details) $ kubectl get secret sample-cluster -o jsonpath="{.data.port}" | base64 --decode
11211
```

**Comment by aiell0** [comment]
/test-examples="examples/elasticache/v1beta1/cluster.yaml"

**Review by turkenf — APPROVED** [review]
Thank you @aiell0, LGTM.

**Pull request merged**